util/stream_dataset.py

```diff
@@ -391,7 +391,7 @@ def _insert_data(dataset, param, data, provenance_metadata=None, request_id=None
             # remove obs dimension from parameter's dimensions and data (13025 AC2)
             param_dimensions.remove('-obs')
             dims = param_dimensions
-            data = data[0]
+            data = data[0] if data else None
```

Comment:
Does data being None have impacts later in the execution path?

Reply:
The rest of the code checks for data being none and handles it appropriately.  This code was something new I added for #13025, I initially had the check in for None data, but re-organized the logic and forgot to put it back in.